# 1.2 Google form analysis tests with sessions

In [167]:
#from datetime import datetime
%run "../Functions/1. Google form analysis.ipynb"
#%run "../Functions/2. Game sessions.ipynb"

C:\Users\Mikael\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2902: DtypeWarning: Columns (18,21,22,28,32,38) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
C:\Users\Mikael\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2902: DtypeWarning: Columns (14,15,16,17,18,20,21,22,25,26,27,30,33,34,35,36,37,38,39,40,41,42,43) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
C:\Users\Mikael\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2902: DtypeWarning: Columns (18,21,22,28,32,38) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
C:\Users\Mikael\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2902: DtypeWarning: Columns (14,15,16,17,18,20,21,22,25,26,27,30,33,34,35,36,37,

(userIDThatDidNotAnswer)
(userID1AnswerEN)
(userIDAnswersEN)
(userID1ScoreEN)
(userIDScoresEN)
(userID1AnswerFR)
(userIDAnswersFR)
(userID1ScoreFR)
(userIDScoresFR)
(userIDAnswersENFR)

#### getAnswerTemporality tinkering

In [168]:
allGFormResponders = getAllResponders()

##### users that played more than once

In [169]:
for uid in allGFormResponders:
    if(len(getAllSessionsOfUser(df152, uid, True)) > 1):
        print(uid)

"7037c5b2-c286-498e-9784-9a061c778609"
"bfdfd356-5d6f-4696-a2f1-c1dc338aa64b"
"a7936587-8b71-43b6-9c61-17b2c2b55de3"
"bffd0877-e77e-470a-b0f9-a9cdc6ff0a09"
"97f6ae4f-e2a6-4721-8936-f759c8867f6a"
"6b5d392d-b737-49ef-99af-e8c445ff6379"
"00dbbdca-d86c-4bc9-803c-0602e0153f68"
"bffd0877-e77e-470a-b0f9-a9cdc6ff0a09"
"d3cf1687-6c27-45e7-ada7-6619ef639dbe"
"a4d4b030-9117-4331-ba48-90dc05a7e65a"
"d6826fd9-a6fc-4046-b974-68e50576183f"
"a4d4b030-9117-4331-ba48-90dc05a7e65a"
"deb089c0-9be3-4b75-9b27-28963c77b10c"
"58d22690-8604-41cf-a5b7-d71fb3b9ad5b"
"24d954e3-23d4-4368-9b10-a1462d2c50d0"
"cad582e3-1868-4c8f-a932-74d7498c9934"
"58d22690-8604-41cf-a5b7-d71fb3b9ad5b"
"24d954e3-23d4-4368-9b10-a1462d2c50d0"
"97f6ae4f-e2a6-4721-8936-f759c8867f6a"
"a7936587-8b71-43b6-9c61-17b2c2b55de3"
"4cd1abad-a077-46a1-bbe4-7a83594de09e"
"4dbc1a0a-e1d2-44dd-833c-62252a56bfd6"
"bfdfd356-5d6f-4696-a2f1-c1dc338aa64b"
"9880ce9d-c6fc-476e-98a9-a9e8c8f7fcdc"
"deb089c0-9be3-4b75-9b27-28963c77b10c"


In [170]:
type(allGFormResponders)

list

In [171]:
len(allGFormResponders)

73

In [172]:
allUniqueGFormResponders = pd.Series(allGFormResponders).unique()
len(allUniqueGFormResponders)

51

##### users that answered more than once

In [173]:
allMultipleGFormResponders = pd.Series(allGFormResponders).value_counts()
allMultipleGFormResponders = allMultipleGFormResponders[allMultipleGFormResponders > 1]
allMultipleGFormResponders = pd.Series(allMultipleGFormResponders.index)
#allMultipleGFormResponders

In [174]:
testUserID = '"bfdfd356-5d6f-4696-a2f1-c1dc338aa64b"' # 4 sessions

In [175]:
#testUserID = '"a7936587-8b71-43b6-9c61-17b2c2b55de3"' # 2 sessions

#### gform timestamps

In [176]:
gformTimestamps = gform[gform[localplayerguidkey]==testUserID.replace('"','')]['Timestamp']

In [177]:
gform[gform['Timestamp'] > '2017/10/15']['Timestamp'].sort_values()

5     2017/10/16 12:46:49 PM GMT+1
41    2017/10/16 12:48:03 PM GMT+1
6     2017/10/16 12:48:34 PM GMT+1
42    2017/10/16 12:48:39 PM GMT+1
43    2017/10/16 12:48:56 PM GMT+1
44    2017/10/16 12:50:18 PM GMT+1
7     2017/10/16 12:50:38 PM GMT+1
8     2017/10/16 12:50:40 PM GMT+1
9     2017/10/16 12:51:49 PM GMT+1
10    2017/10/16 12:51:54 PM GMT+1
11    2017/10/16 12:53:04 PM GMT+1
45    2017/10/16 12:53:58 PM GMT+1
12    2017/10/16 12:54:50 PM GMT+1
13    2017/10/16 12:55:31 PM GMT+1
14    2017/10/16 12:56:10 PM GMT+1
46    2017/10/16 12:56:24 PM GMT+1
15    2017/10/16 12:56:26 PM GMT+1
47    2017/10/16 12:58:10 PM GMT+1
16    2017/10/16 12:58:47 PM GMT+1
17     2017/10/16 1:01:02 PM GMT+1
18     2017/10/16 1:04:27 PM GMT+1
48     2017/10/16 1:04:50 PM GMT+1
49     2017/10/16 1:09:53 PM GMT+1
50     2017/10/16 1:10:58 PM GMT+1
51     2017/10/16 1:12:15 PM GMT+1
52     2017/10/16 1:40:40 PM GMT+1
53     2017/10/16 1:45:58 PM GMT+1
54     2017/10/16 1:55:20 PM GMT+1
55     2017/10/16 1:

In [178]:
pd.to_datetime('2017/10/16 2:14:12 PM GMT+1')

Timestamp('2017-10-16 15:14:12')

In [179]:
pd.to_datetime('2017/10/16 2:14:12 PM GMT+1', utc=True)

Timestamp('2017-10-16 15:14:12+0000', tz='UTC')

In [180]:
pd.to_datetime('2017/10/16 2:14:12 PM')

Timestamp('2017-10-16 14:14:12')

#### RedMetrics timestamps

In [181]:
pd.to_datetime('2017-11-02T11:08:03.813Z')

Timestamp('2017-11-02 11:08:03.813000')

In [182]:
pd.to_datetime('2017-11-02T11:08:03.813Z', utc=True)

Timestamp('2017-11-02 11:08:03.813000+0000', tz='UTC')

In [ ]:
df152\
.where(df152['userTime']>'2017-10-26T12:06')\
.where(df152['userTime']<'2017-10-26T12:07')\
['userTime'].sort_values()
#.dropna(subset=['userTime'])

In [222]:
#df152['2017-10-24T08:15:11' < df152.userTime < '2017-10-24T08:15:22']

In [227]:
df152.query("('2017-10-24T08:15:11' < userTime < '2017-10-24T08:15:22')")

coordinates customData customData.biobrick customData.chapter  \
74653  [-553, -188]        NaN                 NaN                NaN   
78577  [-548, -145]        NaN                 NaN                NaN   
78578  [-548, -145]        NaN                 NaN                NaN   

      customData.controls customData.count         customData.device  \
74653                 NaN              "3"                       NaN   
78577                 NaN              NaN  "PRBAD3:RBS2:FLUO1:DTER"   
78578                 NaN              NaN  "PRCONS:RBS2:FLUO1:DTER"   

                  customData.devices customData.dnabit customData.duration  \
74653                            NaN               NaN                 NaN   
78577  "slot0[PRCONS:RBS1:MOV:DTER]"               NaN                 NaN   
78578  "slot0[PRCONS:RBS1:MOV:DTER]"               NaN                 NaN   

                 ...            playerCountry playerCustomData  \
74653            ...                      NaN              NaN   
78577            ...                      NaN              NaN   
78578            ...                      NaN              NaN   

      playerExternalId playerGender                             sessionId  \
74653              NaN          NaN  eacf1aa8-5072-43fe-a984-62c5ea5bf008   
78577              NaN          NaN  eacf1aa8-5072-43fe-a984-62c5ea5bf008   
78578              NaN          NaN  eacf1aa8-5072-43fe-a984-62c5ea5bf008   

      playerRegion                 section                serverTime    type  \
74653          NaN  tutorial1.Checkpoint04  2017-10-24T08:15:12.779Z  pickup   
78577          NaN  tutorial1.Checkpoint04  2017-10-24T08:15:20.215Z  remove   
78578          NaN  tutorial1.Checkpoint04  2017-10-24T08:15:22.304Z     add   

                       userTime  
74653  2017-10-24T08:15:11.231Z  
78577  2017-10-24T08:15:19.380Z  
78578  2017-10-24T08:15:21.378Z  

[3 rows x 39 columns]

In [228]:
df152[df152['userTime']>'2017-10-16T11:58:03']['userTime'].sort_values().head()

66989    2017-10-16T11:58:03.987Z
66992    2017-10-16T11:58:05.126Z
66990    2017-10-16T11:58:08.845Z
66991    2017-10-16T11:58:09.292Z
66993    2017-10-16T11:58:12.006Z
Name: userTime, dtype: object

#### comparison

firstGameTime='2017-10-16 12:06:39.217000+00:00'
dateGform='2017-10-16 14:04:27+00:00'
dateGform='2017-10-16 15:22:25+00:00'
firstGameTime='2017-10-16 11:58:03.987000+00:00'
dateGform='2017-10-16 13:55:31+00:00'
dateGform='2017-10-16 15:06:32+00:00'

67603    2017-10-16T12:06:39.217Z
18     2017/10/16 1:04:27 PM GMT+1
23     2017/10/16 2:22:25 PM GMT+1

66989    2017-10-16T11:58:03.987Z      
13    2017/10/16 12:55:31 PM GMT+1
59     2017/10/16 2:06:32 PM GMT+1

In [229]:
rmfirst1 = '2017-10-16T12:06:39.217Z'
gfa1before1 = '2017/10/16 1:04:27 PM GMT+1'
gfa1after1 = '2017/10/16 2:22:25 PM GMT+1'

rmfirst2 = '2017-10-16T11:58:03.987Z'      
gfa1before2 = '2017/10/16 12:55:31 PM GMT+1'
gfa1after2 = '2017/10/16 2:06:32 PM GMT+1'

In [230]:
dt_rmfirst1 = pd.to_datetime(rmfirst1)
dt_gfa1before1 = pd.to_datetime(gfa1before1)
dt_gfa1after1 = pd.to_datetime(gfa1after1)

dt_rmfirst2 = pd.to_datetime(rmfirst2)
dt_gfa1before2 = pd.to_datetime(gfa1before2)
dt_gfa1after2 = pd.to_datetime(gfa1after2)

In [231]:
dt_rmfirst1 = pd.to_datetime(rmfirst1)
dt_gfa1before1 = pd.to_datetime(gfa1before1)
dt_gfa1after1 = pd.to_datetime(gfa1after1)

dt_rmfirst2 = pd.to_datetime(rmfirst2)
dt_gfa1before2 = pd.to_datetime(gfa1before2)
dt_gfa1after2 = pd.to_datetime(gfa1after2)

In [232]:
dt_rmfirst1utc = pd.to_datetime(rmfirst1, utc=True)
dt_gfa1before1utc = pd.to_datetime(gfa1before1).tz_localize('Europe/Berlin')
dt_gfa1after1utc = pd.to_datetime(gfa1after1).tz_localize('Europe/Berlin')

dt_rmfirst2utc = pd.to_datetime(rmfirst2, utc=True)
dt_gfa1before2utc = pd.to_datetime(gfa1before2).tz_localize('Europe/Berlin')
dt_gfa1after2utc = pd.to_datetime(gfa1after2).tz_localize('Europe/Berlin')

In [233]:
[dt_rmfirst1,dt_gfa1before1,dt_gfa1after1,dt_rmfirst2,dt_gfa1before2,dt_gfa1after2]

[Timestamp('2017-10-16 12:06:39.217000'),
 Timestamp('2017-10-16 14:04:27'),
 Timestamp('2017-10-16 15:22:25'),
 Timestamp('2017-10-16 11:58:03.987000'),
 Timestamp('2017-10-16 13:55:31'),
 Timestamp('2017-10-16 15:06:32')]

In [234]:
[dt_rmfirst1utc,dt_gfa1before1utc,dt_gfa1after1utc,dt_rmfirst2utc,dt_gfa1before2utc,dt_gfa1after2utc]

[Timestamp('2017-10-16 12:06:39.217000+0000', tz='UTC'),
 Timestamp('2017-10-16 14:04:27+0200', tz='Europe/Berlin'),
 Timestamp('2017-10-16 15:22:25+0200', tz='Europe/Berlin'),
 Timestamp('2017-10-16 11:58:03.987000+0000', tz='UTC'),
 Timestamp('2017-10-16 13:55:31+0200', tz='Europe/Berlin'),
 Timestamp('2017-10-16 15:06:32+0200', tz='Europe/Berlin')]

In [235]:
[dt_rmfirst1utc>dt_gfa1before1utc,dt_rmfirst1utc<dt_gfa1after1utc,dt_rmfirst2utc>dt_gfa1before2utc,dt_rmfirst2utc<dt_gfa1after2utc]

[True, True, True, True]

In [236]:
stamp = pd.to_datetime('2017-11-02T11:08:03.813Z', utc=True)
stamp

Timestamp('2017-11-02 11:08:03.813000+0000', tz='UTC')

In [237]:
stamp = pd.to_datetime('2017-11-02T11:08:03.813Z')
type(stamp)

pandas._libs.tslib.Timestamp

In [238]:
#stamp = stamp.tz_localize('America/Sao_Paulo')
stamp = stamp.tz_localize('GMT')
new_stamp = stamp.tz_convert('US/Eastern')
new_stamp

Timestamp('2017-11-02 07:08:03.813000-0400', tz='US/Eastern')

In [239]:
stamp

Timestamp('2017-11-02 11:08:03.813000+0000', tz='GMT')

In [240]:
new_stamp

Timestamp('2017-11-02 07:08:03.813000-0400', tz='US/Eastern')

In [241]:
for timestamp in gformTimestamps:
    print(timestamp)

2017/10/16 12:50:40 PM GMT+1
2017/10/16 2:14:16 PM GMT+1


In [242]:
sessions = getAllSessionsOfUser(df152, testUserID, True)
len(sessions)

4

In [243]:
sessions['sessionId']

66605    e9aa70e2-3a0c-4742-a5fd-489d4902fdf8
66699    d641e7d8-7276-403a-a53b-f7622fd2153d
66734    72c84ce0-f5da-4925-be3e-344675516ed9
67057    3f3c09d8-a13a-4c3d-9044-8af37a46508c
Name: sessionId, dtype: object

In [244]:
firstGameTime = ''
lastGameTime = ''
for session in sessions['sessionId']:
    timedEvents = df152[df152['sessionId']==session]
    
    earliest = timedEvents['userTime'].min()
    if('' == firstGameTime or (earliest < firstGameTime)):
        firstGameTime = earliest
    latest = timedEvents['userTime'].max()
    if('' == lastGameTime or (latest > lastGameTime)):
        lastGameTime = latest
        
firstGameTime = pd.to_datetime(firstGameTime)
lastGameTime = pd.to_datetime(lastGameTime)
print("firstGameTime=" + str(firstGameTime))
print("lastGameTime=" + str(lastGameTime))

firstGameTime=2017-10-16 11:42:46.503000
lastGameTime=2017-10-16 13:01:39.388000


In [245]:
firstGameTime = datetime.max
lastGameTime = datetime.min


for session in sessions['sessionId']:
    timedEvents = df152[df152['sessionId']==session]
    
    timedEvents['userTime'] = timedEvents['userTime'].map(pd.to_datetime)
    #print(timedEvents['userTime'])
    
    earliest = timedEvents['userTime'].min()
    if(earliest < firstGameTime):
        firstGameTime = earliest
        
    latest = timedEvents['userTime'].max()
    if(latest > lastGameTime):
        lastGameTime = latest
        
firstGameTime = pd.to_datetime(firstGameTime)
lastGameTime = pd.to_datetime(lastGameTime)
print("firstGameTime=" + str(firstGameTime))
print("lastGameTime=" + str(lastGameTime))

firstGameTime=2017-10-16 11:42:46.503000
lastGameTime=2017-10-16 13:01:39.388000


C:\Users\Mikael\Anaconda3\lib\site-packages\ipykernel\__main__.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy


In [246]:
for timestamp in gformTimestamps:
    dateGform = pd.to_datetime(timestamp)
    print(dateGform)
    if(dateGform < firstGameTime):
        print(answerTemporalities[0])
    elif (dateGform > lastGameTime):
        print(answerTemporalities[1])
    else:
        print(answerTemporalities[2])

2017-10-16 13:50:40
after
2017-10-16 15:14:16
after


In [247]:
df152.columns

Index(['coordinates', 'customData', 'customData.biobrick',
       'customData.chapter', 'customData.controls', 'customData.count',
       'customData.device', 'customData.devices', 'customData.dnabit',
       'customData.duration', 'customData.energy', 'customData.graphics',
       'customData.language', 'customData.life', 'userId',
       'customData.message', 'customData.nanobot', 'customData.newtab',
       'customData.option', 'customData.plasmid', 'customData.platform',
       'customData.sametab', 'customData.slot', 'customData.sound',
       'customData.source', 'customData.total', 'gameVersion', 'id',
       'playerBirthdate', 'playerCountry', 'playerCustomData',
       'playerExternalId', 'playerGender', 'sessionId', 'playerRegion',
       'section', 'serverTime', 'type', 'userTime'],
      dtype='object')

In [248]:
#df152['section']
#df152.dropna(subset=['section'])['section']

In [249]:
earliest

Timestamp('2017-10-16 11:59:30.726000')

In [250]:
for userId in allMultipleGFormResponders:
    firstGameTime = getFirstEventDate(getRedMetricsGUIDFormat(userId))
    
    strDebug = '\n\n\nuid='+str(userId)+'\n=' + str(firstGameTime)
    
    gformTimestamps = gform[gform[localplayerguidkey]==userId.replace('"','')]['Timestamp']
    
    for timestamp in gformTimestamps:
        # Google forms Timestamps are GMT+1
        dateGform = pd.to_datetime(timestamp).tz_localize('Europe/Berlin')
        #print('dateGform='+str(dateGform))
        
        if(dateGform < firstGameTime):
            strDebug = strDebug + '\n' + str(dateGform) + '=>' + str(answerTemporalities[0])
        elif (dateGform > firstGameTime):
            strDebug = strDebug + '\n' + str(dateGform) + '=>' + str(answerTemporalities[1])
        else:
            strDebug = strDebug + '\n' + str(dateGform) + '=>' + str(answerTemporalities[2])
            
    print(strDebug)




uid="deb089c0-9be3-4b75-9b27-28963c77b10c"
=2017-06-07 11:42:02.641000+00:00
2017-06-09 18:54:51+02:00=>after
2017-11-14 09:10:15+01:00=>after



uid="01e85778-2903-447b-bbab-dd750564ee2d"
=2017-10-12 13:07:18.652000+00:00
2017-10-12 15:05:21+02:00=>before
2017-10-12 16:15:02+02:00=>after



uid="5ecf601d-4eac-433e-8056-3a5b9eda0555"
=2017-10-16 12:06:39.217000+00:00
2017-10-16 14:04:27+02:00=>before
2017-10-16 15:22:25+02:00=>after



uid="97f6ae4f-e2a6-4721-8936-f759c8867f6a"
=2017-10-16 11:57:23.734000+00:00
2017-10-16 13:56:26+02:00=>before
2017-10-16 14:57:32+02:00=>after



uid="5977184a-1be2-4725-9b48-f2782dc03efb"
=2017-10-16 11:55:19.480000+00:00
2017-10-16 13:53:04+02:00=>before
2017-10-16 15:14:17+02:00=>after



uid="a7936587-8b71-43b6-9c61-17b2c2b55de3"
=2017-10-16 11:57:42.143000+00:00
2017-10-16 13:51:49+02:00=>before
2017-10-16 15:07:42+02:00=>after



uid="bfdfd356-5d6f-4696-a2f1-c1dc338aa64b"
=2017-10-16 11:56:19.399000+00:00
2017-10-16 13:50:40+02:00=>before
2017-

In [251]:
gform[localplayerguidkey][12]

'e63a2fc6-50bd-4417-a712-fe688bf8e242'

In [252]:
gform[localplayerguidkey][50]

'a9f9efd2-b6bf-4bda-97d1-3620800bd62f'

In [253]:
getGFormGUIDFormat( gform[localplayerguidkey][50])

'a9f9efd2-b6bf-4bda-97d1-3620800bd62f'

In [254]:
getFirstEventDate(getGFormGUIDFormat( gform[localplayerguidkey][50]) )

Timestamp('2262-04-11 23:47:16.854775807+0000', tz='UTC')

In [255]:
getAllSessionsOfUser(df152, gform[localplayerguidkey][3], True)

Empty DataFrame
Columns: [userId, sessionId]
Index: []

In [256]:
#for index in gform.index:
#    print(index)

In [257]:
gform['Timestamp'][0]

'2017/08/18 3:22:26 PM GMT+1'

In [258]:
gform.loc[0][localplayerguidkey]

'8d352896-a3f1-471c-8439-0f426df901c1'

In [259]:
gform

Timestamp Are you interested in video games?  \
0    2017/08/18 3:22:26 PM GMT+1                          Extremely   
1    2017/09/06 1:28:03 PM GMT+1                         Moderately   
2    2017/09/12 9:02:06 AM GMT+1                              A lot   
3    2017/09/13 2:11:34 AM GMT+1                       I don't know   
4    2017/09/17 6:34:56 PM GMT+1                         Moderately   
5   2017/10/16 12:46:49 PM GMT+1                         Moderately   
6   2017/10/16 12:48:34 PM GMT+1                              A lot   
7   2017/10/16 12:50:38 PM GMT+1                         Moderately   
8   2017/10/16 12:50:40 PM GMT+1                         Moderately   
9   2017/10/16 12:51:49 PM GMT+1                              A lot   
10  2017/10/16 12:51:54 PM GMT+1                         Moderately   
11  2017/10/16 12:53:04 PM GMT+1                           Slightly   
12  2017/10/16 12:54:50 PM GMT+1                         Moderately   
13  2017/10/16 12:55:31 PM GMT+1                              A lot   
14  2017/10/16 12:56:10 PM GMT+1                              A lot   
15  2017/10/16 12:56:26 PM GMT+1                          Extremely   
16  2017/10/16 12:58:47 PM GMT+1                           Slightly   
17   2017/10/16 1:01:02 PM GMT+1                          Extremely   
18   2017/10/16 1:04:27 PM GMT+1                           Slightly   
19   2017/10/16 2:07:57 PM GMT+1                              A lot   
20   2017/10/16 2:11:47 PM GMT+1                         Moderately   
21   2017/10/16 2:14:17 PM GMT+1                           Slightly   
22   2017/10/16 2:18:50 PM GMT+1                           Slightly   
23   2017/10/16 2:22:25 PM GMT+1                           Slightly   
24   2017/10/16 2:22:44 PM GMT+1                              A lot   
25  2017/10/19 11:57:15 AM GMT+1                              A lot   
26   2017/10/22 9:00:20 AM GMT+1                              A lot   
27  2017/10/25 10:29:57 PM GMT+1                              A lot   
28   2017/10/26 3:44:06 PM GMT+1                              A lot   
29   2017/11/12 6:04:19 PM GMT+1                         Moderately   
..                           ...                                ...   
43  2017/10/16 12:48:56 PM GMT+1                           Beaucoup   
44  2017/10/16 12:50:18 PM GMT+1                        Moyennement   
45  2017/10/16 12:53:58 PM GMT+1                         Enormément   
46  2017/10/16 12:56:24 PM GMT+1                        Moyennement   
47  2017/10/16 12:58:10 PM GMT+1                        Moyennement   
48   2017/10/16 1:04:50 PM GMT+1                        Moyennement   
49   2017/10/16 1:09:53 PM GMT+1                             Un peu   
50   2017/10/16 1:10:58 PM GMT+1                        Moyennement   
51   2017/10/16 1:12:15 PM GMT+1                           Beaucoup   
52   2017/10/16 1:40:40 PM GMT+1                        Moyennement   
53   2017/10/16 1:45:58 PM GMT+1                         Enormément   
54   2017/10/16 1:55:20 PM GMT+1                        Moyennement   
55   2017/10/16 1:57:32 PM GMT+1                         Enormément   
56   2017/10/16 2:04:04 PM GMT+1                           Beaucoup   
57   2017/10/16 2:04:12 PM GMT+1                        Moyennement   
58   2017/10/16 2:05:35 PM GMT+1                        Moyennement   
59   2017/10/16 2:06:32 PM GMT+1                         Enormément   
60   2017/10/16 2:07:00 PM GMT+1                        Moyennement   
61   2017/10/16 2:07:42 PM GMT+1                        Moyennement   
62   2017/10/16 2:11:08 PM GMT+1                        Moyennement   
63   2017/10/16 2:12:27 PM GMT+1                        Moyennement   
64   2017/10/16 2:14:12 PM GMT+1                        Moyennement   
65   2017/10/16 2:14:16 PM GMT+1                        Moyennement   
66   2017/10/16 2:33:40 PM GMT+1                             Un peu   
67  2017/10/20 11:45:27 AM GMT+1               

In [260]:
#from datetime import datetime
%run "../Functions/1. Google form analysis.ipynb"
#%run "../Functions/2. Game sessions.ipynb"

C:\Users\Mikael\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2902: DtypeWarning: Columns (18,21,22,28,32,38) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
C:\Users\Mikael\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2902: DtypeWarning: Columns (14,15,16,17,18,20,21,22,25,26,27,30,33,34,35,36,37,38,39,40,41,42,43) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
C:\Users\Mikael\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2902: DtypeWarning: Columns (18,21,22,28,32,38) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
C:\Users\Mikael\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2902: DtypeWarning: Columns (14,15,16,17,18,20,21,22,25,26,27,30,33,34,35,36,37,

In [261]:
gform['Temporality'] = gform['Timestamp']
for index in gform.index:
    firstEventDate = getFirstEventDate(getRedMetricsGUIDFormat(gform[localplayerguidkey][index]))
    gform['Temporality'][index] = getTemporality(gform['Timestamp'][index],firstEventDate)
    
    #print('uid=' + str(gform.loc[index][localplayerguidkey]) + ': ' \
    #      + str(firstEventDate) + ' & ' \
    #      + str(gform['Timestamp'][index]) + ' -> ' + str(gform['Temporality'][index]))
print(gform['Temporality'].unique())
print(gform['Temporality'].value_counts())

['after' 'undefined' 'before']
after        37
before       26
undefined    10
Name: Temporality, dtype: int64


C:\Users\Mikael\Anaconda3\lib\site-packages\ipykernel\__main__.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy


In [262]:
print(gform['Temporality'][0])

after


In [263]:
print(gform.loc[index,'Temporality'])

after


In [264]:
gform['Temporality'] = gform['Timestamp'].copy()
for index in gform.index:
    firstEventDate = getFirstEventDate(getRedMetricsGUIDFormat(gform.loc[index,localplayerguidkey]))
    #print(gform.loc[index,'Temporality'])
    #print(gform['Timestamp'][index])
    #print(getTemporality(gform['Timestamp'][index],firstEventDate))
    gform.loc[index,'Temporality'] = \
    getTemporality(gform.loc[index,'Timestamp'],firstEventDate)
print(gform['Temporality'].unique())
print(gform['Temporality'].value_counts())

['after' 'undefined' 'before']
after        37
before       26
undefined    10
Name: Temporality, dtype: int64


In [265]:
setAnswerTemporalities(gform)

In [266]:
print(gform['Temporality'].unique())
print(gform['Temporality'].value_counts())

['after' 'undefined' 'before']
after        37
before       26
undefined    10
Name: Temporality, dtype: int64


#### getRedMetricsGUIDFormat & getGFormGUIDFormat tinkering

In [267]:
gform[localplayerguidkey][0]

'8d352896-a3f1-471c-8439-0f426df901c1'

In [268]:
#df152['userId'].dropna().unique()

In [269]:
pattern1 = '[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}'
pattern2 = '"' + pattern1 + '"'
pattern3 = '^' + pattern1 + '$'
pattern4 = '^' + pattern2 + '$'
patterns = [pattern1, pattern2, pattern3, pattern4]

ids = [gform[localplayerguidkey][0], df152['userId'][0]]

for id in ids:
    for pattern in patterns:
        res = re.search(pattern, id)
        if(res):
            print("match\n" + id + "\n" + pattern + "\n\n\n")
        else:
            print("no match\n" + id + "\n" + pattern + "\n\n\n")

match
8d352896-a3f1-471c-8439-0f426df901c1
[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}



no match
8d352896-a3f1-471c-8439-0f426df901c1
"[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}"



match
8d352896-a3f1-471c-8439-0f426df901c1
^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$



no match
8d352896-a3f1-471c-8439-0f426df901c1
^"[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}"$



match
"4dbc2f43-421c-4e23-85d4-f17723ff8c66"
[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}



match
"4dbc2f43-421c-4e23-85d4-f17723ff8c66"
"[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}"



no match
"4dbc2f43-421c-4e23-85d4-f17723ff8c66"
^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$



match
"4dbc2f43-421c-4e23-85d4-f17723ff8c66"
^"[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}"$





In [270]:
#from datetime import datetime
%run "../Functions/1. Google form analysis.ipynb"
#%run "../Functions/2. Game sessions.ipynb"

C:\Users\Mikael\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2902: DtypeWarning: Columns (18,21,22,28,32,38) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
C:\Users\Mikael\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2902: DtypeWarning: Columns (14,15,16,17,18,20,21,22,25,26,27,30,33,34,35,36,37,38,39,40,41,42,43) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
C:\Users\Mikael\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2902: DtypeWarning: Columns (18,21,22,28,32,38) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)
C:\Users\Mikael\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2902: DtypeWarning: Columns (14,15,16,17,18,20,21,22,25,26,27,30,33,34,35,36,37,

In [271]:
getRedMetricsGUIDFormat( gform[localplayerguidkey][0] )

'"8d352896-a3f1-471c-8439-0f426df901c1"'

In [272]:
getRedMetricsGUIDFormat( df152['userId'][0] )

'"4dbc2f43-421c-4e23-85d4-f17723ff8c66"'

In [273]:
getGFormGUIDFormat( gform[localplayerguidkey][0] )

'8d352896-a3f1-471c-8439-0f426df901c1'

In [274]:
getGFormGUIDFormat( df152['userId'][0] )

'4dbc2f43-421c-4e23-85d4-f17723ff8c66'

In [275]:
df152['userId'][0]

'"4dbc2f43-421c-4e23-85d4-f17723ff8c66"'

In [276]:
print(re.search(RedMetricsGUIDpattern, df152['userId'][0]))

<_sre.SRE_Match object; span=(0, 38), match='"4dbc2f43-421c-4e23-85d4-f17723ff8c66"'>


In [277]:
gform[gform[localplayerguidkey]=='0bb50c7b-d9c4-466d-b906-84e46896f41d']

Empty DataFrame
Columns: [Timestamp, Are you interested in video games?, Do you play video games?, How old are you?, What is your gender?, How long have you studied biology?, Are you interested in biology?, Before playing Hero.Coli, had you ever heard about synthetic biology?, Before playing Hero.Coli, had you ever heard about BioBricks?, Have you ever played an older version of Hero.Coli before?, Have you played the current version of Hero.Coli?, Have you played the arcade cabinet version of Hero.Coli?, Have you played the Android version of Hero.Coli?, In order to modify the abilities of the bacterium, you have to..., What are BioBricks and devices?, What is the name of this BioBrick?, What is the name of this BioBrick?.1, What is the name of this BioBrick?.2, What is the name of this BioBrick?.3, What does this BioBrick do?, What does this BioBrick do?.1, What does this BioBrick do?.2, What does this BioBrick do?.3, Pick the case where the BioBricks are well-ordered:, When does green fluorescence happen?, What happens when you unequip the movement device?, What is this?, What does this device do?, What does this device do?.1, What does this device do?.2, What does this device do?.3, What does this device do?.4, What does this device do?.5, What does this device do?.6, What does this device do?.7, Guess: what would a device producing l-arabinose do, if it started with a l-arabinose-induced promoter?, Guess: the bacterium would glow yellow..., What is the species of the bacterium of the game?, What is the scientific name of the tails of the bacterium?, Find the antibiotic:, You can write down remarks here., Do not edit -  pre-filled anonymous ID, Language]
Index: []

[0 rows x 43 columns]